### PR TITLE
r.geomorphon: separate input from output

### PR DESCRIPTION
**Pull request by infrastation**
These three commits remove some excess code in a way that should be easy to review. It would be fine to squash these changes together.

### CODE REVIEW

**Comment by infrastation** [comment]
One of the checks had timed out after 6 hours.

File: raster/r.geomorphon/main.c [hunk 1]
```diff
@@ -117,11 +112,11 @@ int main(int argc, char **argv)
 	G_add_keyword(_("machine vision geomorphometry"));
 
 	opt_input = G_define_standard_option(G_OPT_R_ELEV);
-	opt_input->key = rasters[0].name;
+	opt_input->key = "elevation";
 	opt_input->required = YES;
-	opt_input->description = _(rasters[0].description);
+	opt_input->description = _("Name of input elevation raster map");
```
Comment (wenzeslaus) [inline raster/r.geomorphon/main.c]: You can actually drop key and description here. See https://grass.osgeo.org/grass78/manuals/parser_standard_options.html

Comment (infrastation) [reply raster/r.geomorphon/main.c]: You are right. And `required` is excess as well.

File: raster/r.geomorphon/main.c [hunk 2]
```diff
@@ -122,7 +122,7 @@ int main(int argc, char **argv)
 	opt_input->required = rasters[0].required;
 	opt_input->description = _(rasters[0].description);
 
-	for (i = 1; i < io_size; ++i) {	/* WARNING: loop starts from one, zero is for input */
+	for (i = o_forms; i < io_size; ++i) {	/* WARNING: loop starts from one, zero is for input */
```
[the remark this answers is not in the document]
Comment (infrastation) [reply raster/r.geomorphon/main.c]: The comment at the first for loop says that, but not at the subsequent loops. `opt_output[0]` is accessed outside of the for loops, although to a questionable effect, so the next commit removes the element from the array. Let me push an updated version to see.

[the remark this answers is not in the document]
Comment (wenzeslaus) [reply raster/r.geomorphon/main.c]: Thanks, I didn't mean to submit the review yet and meant to delete this comment. The subsequent commit resolved this. I don't know if you needed to update anything.

File: raster/r.geomorphon/main.c (hunk 1 above)
Comment (wenzeslaus) [reply raster/r.geomorphon/main.c]: Right, esp. because elevation is the only input, so there is no need to be explicit about elevation being required.

**Review by wenzeslaus — APPROVED** [review]
Shorter and simpler code. What else to ask for?

I guess the removal of typedef for IO rasters is a matter of taste. It makes sense to me here. It might be less readable for people coming from Python, but that's hard to guess.

**Comment by infrastation** [comment]
It would be great to have this merged so I can rebase and open the next pull request with the remaining code cleanups.

**Comment by neteler** [comment]
I take liberty to merge it

**Pull request merged**